### ISSUE TITLE
3.0 - unneeded payment_meta is still migrated

### ISSUE THREAD

**Issue by robincornett** [easydigitaldownloads/easy-digital-downloads#8603]
### Expected behavior
When I migrate my site data from 2.10 to 3.0, I expect to only have `payment_meta` saved to the new orders if there is something unique (for example, from an extension) in the metadata which needs to be preserved.

### Actual behavior
When I run the migration, some [core metadata](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/admin/upgrades/v3/class-data-migrator.php#L739-L750) is retained. This may be because I'm looking at a data set for a fairly old site (another migration doesn't have all of these), but I'm seeing a lot of entries with:
- `amount`
- `user_id`

### Steps to reproduce the behavior
1. Run a migration from EDD 2.10 to 3.0. I'm going to guess the data needs to be from a fairly old site.
2. Check the `edd_ordermeta` table for `payment_meta` key/values with these two items.

### Information (if a specific version is affected):
PHP Version: 7.3.5
EDD Version (or branch): release/3.0
WordPress Version: 5.7.1

**Comment by robincornett** [comment]
@ashleyfae I think we can actually close this--`amount` and `user_id` are now fine.

However, it might be nice to add a filter on the core payment meta array, the way we did on the user info, so that extensions could append to that to save a step in the migration process (when I worked on the user history extension ([PR](https://github.com/easydigitaldownloads/edd-user-history/pull/26)), I added steps to update/delete the payment meta once the user history was migrated, but with a filter in place I would not have to do that). This could be a second PR on this issue or I could make a new issue for it.
